manifests/agent/windows.pp

```diff
@@ -1,73 +1,71 @@
-class distelli::agent::windows (
-  $version = '3.66.33',
-  $tempdir = 'C:/Users/distelli/AppData/Local/Temp/1',
-){
+# Install the Puppet Pipelines agent on Windows
+#
+# Do not use directly; use pipelines::agent.
+class pipelines::agent::windows {
+  $env = $facts['pipelines_env']
 
-  if $::facts['os']['hardware'] == 'x86_64' {
-    $archive = "distelli.Windows-AMD64-${version}.gz"
-    $url = "https://s3.amazonaws.com/download.distelli.com/distelli.Windows-AMD64/${archive}"
-  }
-  else {
-    $archive = "distelli.Windows-x86-${version}.gz"
-    $url = "https://s3.amazonaws.com/download.distelli.com/distelli.Windows-x86/${archive}"
-  }
-
-  $workdir = 'C:/Program Files/Distelli'
-
-  archive { "${tempdir}/${archive}" :
-    source       => $url,
-    # cleanup      => false,
-    creates      => "${tempdir}/distelli",
-    extract      => true,
-    extract_path => $tempdir,
-    require      => User['distelli'],
+  if ! $pipelines::agent::install_dir {
+    $install_dir = "${env['ProgramFiles']}\\Distelli"
+  } else {
+    $install_dir = $pipelines::agent::install_dir
   }
 
-  file { $workdir :
-    ensure  => directory,
-    owner   => 'distelli',
-    group   => 'Administrators',
-    require => User['distelli'],
+  if $pipelines::agent::version {
+    $download_url = "${pipelines::agent::download_url}/${pipelines::agent::version}.ps1"
+  } else {
+    $download_url = "${pipelines::agent::download_url}.ps1"
   }
+  $download_location = "${install_dir}\\distelli-download.ps1"
+  $download_cmd = "cmd.exe /c \"powershell -NoProfile -ExecutionPolicy Bypass -Command - < \"${download_location}\"\""
```

Comment:
I might have missed it, but doesn't look like this is being used? But maybe that's because you moved to `provider => powershell`. Note that I'd advise against adding that module dependency if you can help it (and it looks like for your use cases, you can avoid the extra dep)

Reply:
I've deleted the separate $download_cmd. Thanks!